api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -998,18 +999,21 @@ public void markDirty(ResourceType resourceType, long resourceId) {
   }
 
   @Override
-  public DataDictionaryEntry findDataDictionaryEntry(String fieldName, DbCdrVersion cdrVersion) {
-    List<DbDataDictionaryEntry> dataDictionaryEntries =
-        dataDictionaryEntryDao.findByFieldNameAndCdrVersion(fieldName, cdrVersion);
-
-    if (dataDictionaryEntries.isEmpty()) {
+  public DataDictionaryEntry findDataDictionaryEntry(
+      String fieldName, String domain, DbCdrVersion cdrVersion) {
+    CdrVersionContext.setCdrVersionNoCheckAuthDomain(cdrVersion);
+    DbDSDataDictionary dbDSDataDictionary =
+        dsDataDictionaryDao.findDbDSDataDictionaryByFieldNameAndDomain(fieldName, domain);
+    if (dbDSDataDictionary == null) {
       throw new NotFoundException(
           "No Data Dictionary Entry found for domain: "
               + fieldName
               + " cdr version: "
               + cdrVersion);
     }
-    return dataSetMapper.dbModelToClient(dataDictionaryEntries.get(0));
+    DataDictionaryEntry dataDictionaryEntry = dataSetMapper.dbDsModelToClient(dbDSDataDictionary);
+    dataDictionaryEntry.setCdrVersionId(cdrVersion.getCdrVersionId());
```

Comment:
Can we refactor  this to remove the cdrVersionId property from the DataDictionaryEntry client model? The UI should get the cdrVersionId from the workspace and pass that in the API call.

Reply:
Snce cdrVersion and other attributes like definedTime or getTransformedByRegisteredTierPrivacyMethods were not needed in UI, hence deleted it from the Data Object